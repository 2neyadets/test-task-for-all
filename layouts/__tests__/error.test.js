import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import ErrorLayout from '../error'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('ErrorLayout component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(ErrorLayout, {
      localVue,
      stubs: ['NuxtLink'],
      propsData: {
        error: {
          statusCode: 404,
        },
      }
    })
  })

  it('mounts without errors', done => {
    localVue.config.errorHandler = done // для отлавливания ошибок внутри nextTick, если нужно что-то выпонять внутри него
    expect.assertions(1)
    expect(wrapper).toBeTruthy()
    done()
  })

  it('can show corresponding messages', async () => {
    expect.assertions(2)
    expect(wrapper.find('H1').html()).toContain('404')
    wrapper.setProps({
      error: {
        statusCode: undefined,
      },
    })
    await localVue.nextTick()
    expect(wrapper.find('H1').html()).not.toContain('404')
  })

  it('match snapshot', () => {
    expect.assertions(1)
    expect(wrapper.element).toMatchSnapshot()
  })
})
