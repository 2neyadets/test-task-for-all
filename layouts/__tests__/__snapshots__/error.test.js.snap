// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorLayout component match snapshot 1`] = `
<div
  class="full-width full-height flex flex-center"
>
  <div
    class="row"
  >
    <div
      class="col-12 text-center"
    >
      <h1
        class="my-none"
      >
        404
      </h1>
      <h5>
        Страница не найдена
      </h5>
    </div>
    <div
      class="col-12"
    >
      <div
        class="px-md py-sm flex flex-center"
      >
        <nuxtlink-stub
          to="/"
        >
          <btn-stub
            icon=""
            label="На главную"
            size="20"
            type="button"
          />
        </nuxtlink-stub>
      </div>
    </div>
  </div>
</div>
`;
